### ISSUE TITLE
Global variables

### ISSUE THREAD

**Issue by bergtwvd** [docker/compose#3495]
Currently docker compose (1.7.1) does not support global env variables in a compose file. Certain env variables apply to all services and I need to repeat them for every service. It would be good to have an environment section similar to networks and volumes where I can set env variables for all services.

**Comment by mmacfadden** [comment]
Hello, I would like to vote for this as well.  I have a few settings that need to be applied to each container.  I have to repeat them about 8 times in the compose file, which is cumbersome when a developer needs to change one.  It would be great of we could set something like global variables at the top of the file.

Or better yet, if we had a global env_file that provided variables that could then be substituted just like the per container env_file.

**Comment by kaime** [comment]
Allowing to define services which don't have "either an image or a build context specified" would also do the trick:

~~~ yml
version: "2"
services:
  base: 
    env_file: .env
  flask:
    extends: base
    image: python:3.5
    entrypoint: "python app.py"
  nginx:
    extends: base
    image: nginx:alpine
  proxy:
    extends: nginx
    links:
      - flask
~~~

**Comment by shin-** [comment]
Fixed by #5140
